core/deployment/src/main/java/io/quarkus/deployment/ConstructorPropertiesProcessor.java

```diff
@@ -0,0 +1,38 @@
+package io.quarkus.deployment;
+
+import java.beans.ConstructorProperties;
+
+import org.jboss.jandex.AnnotationInstance;
+import org.jboss.jandex.ClassInfo;
+import org.jboss.jandex.DotName;
+import org.jboss.jandex.MethodInfo;
+
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.CombinedIndexBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+
+/**
+ * Registers all classes for reflection,
+ * that contain a constructor annotated with {@link ConstructorProperties}.
+ */
+public class ConstructorPropertiesProcessor {
+
+    private static final DotName CONSTRUCTOR_PROPERTIES = DotName.createSimple(ConstructorProperties.class.getName());
+
+    @BuildStep
+    void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass, CombinedIndexBuildItem indexBuildItem) {
+        indexBuildItem.getIndex().getAnnotations(CONSTRUCTOR_PROPERTIES).stream()
+                .map(AnnotationInstance::target)
+                .filter(MethodInfo.class::isInstance)
+                .map(MethodInfo.class::cast)
+                .map(MethodInfo::declaringClass)
+                .map(ClassInfo::toString)
+                .map(this::asReflectiveClassBuildItem)
+                .forEach(reflectiveClass::produce);
```

Comment:
Honestly, this is far less readable than a good old for loop.

Reply:
I'm kinda used to it, but it'll be indeed more readable using common known technique.